translator/internaldata/oc_to_metrics.go

```diff
@@ -0,0 +1,347 @@
+			// has exact same conditions as we have here in this loop.
+			ocMetricToInternal(ocMetric, combinedMetrics.At(combinedMetricIdx))
+			combinedMetricIdx++
+		} else {
+			// This metric has a different Resource and must be placed in a different
+			// ResourceMetrics instance. Create a separate ResourceMetrics item just for this metric
+			// and store at resourceMetricIdx.
+			ocMetricToResourceMetrics(ocMetric, md.Node, rms.At(resourceMetricIdx))
+			resourceMetricIdx++
+		}
+	}
+
+	return metricData
+}
+
+func ocMetricToResourceMetrics(ocMetric *ocmetrics.Metric, node *occommon.Node, out data.ResourceMetrics) {
+	ocNodeResourceToInternal(node, ocMetric.Resource, out.Resource())
+	ilms := out.InstrumentationLibraryMetrics()
+	ilms.Resize(1)
+	metrics := ilms.At(0).Metrics()
+	metrics.Resize(1)
+	ocMetricToInternal(ocMetric, metrics.At(0))
+}
+
+// ocMetricToInternal conversts ocMetric to internal representation and fill metric
+func ocMetricToInternal(ocMetric *ocmetrics.Metric, metric data.Metric) {
+	metricDescriptor := metric.MetricDescriptor()
+	metricDescriptor.InitEmpty()
+	descriptorToInternal(ocMetric.GetMetricDescriptor(), metricDescriptor)
+	setDataPoints(ocMetric, metric)
+}
+
+func descriptorToInternal(ocDescriptor *ocmetrics.MetricDescriptor, descriptor data.MetricDescriptor) {
+	if ocDescriptor == nil {
+		return
+	}
+
+	descriptorType := descriptorTypeToInternal(ocDescriptor.Type)
+	if descriptorType == invalidMetricType {
+		return
+	}
+
+	descriptor.SetType(descriptorType)
+	descriptor.SetDescription(ocDescriptor.GetDescription())
+	descriptor.SetName(ocDescriptor.GetName())
+	descriptor.SetUnit(ocDescriptor.GetUnit())
+}
+
+func descriptorTypeToInternal(t ocmetrics.MetricDescriptor_Type) data.MetricType {
+	switch t {
+	case ocmetrics.MetricDescriptor_UNSPECIFIED:
+		return data.MetricTypeUnspecified
+	case ocmetrics.MetricDescriptor_GAUGE_INT64:
+		return data.MetricTypeGaugeInt64
+	case ocmetrics.MetricDescriptor_GAUGE_DOUBLE:
+		return data.MetricTypeGaugeDouble
+	case ocmetrics.MetricDescriptor_GAUGE_DISTRIBUTION:
+		return data.MetricTypeGaugeHistogram
+	case ocmetrics.MetricDescriptor_CUMULATIVE_INT64:
+		return data.MetricTypeCounterInt64
+	case ocmetrics.MetricDescriptor_CUMULATIVE_DOUBLE:
+		return data.MetricTypeCounterDouble
+	case ocmetrics.MetricDescriptor_CUMULATIVE_DISTRIBUTION:
+		return data.MetricTypeCumulativeHistogram
+	case ocmetrics.MetricDescriptor_SUMMARY:
+		return data.MetricTypeSummary
+	default:
+		return invalidMetricType
+	}
+}
+
+// setDataPoints converts OC timeseries to internal datapoints based on metric type
+func setDataPoints(ocMetric *ocmetrics.Metric, metric data.Metric) {
+	var int64DataPointsNum, doubleDataPointsNum, histogramDataPointsNum, summaryDataPointsNum int
+	ocLabelsKeys := ocMetric.GetMetricDescriptor().GetLabelKeys()
+	ocPointsCount := getPointsCount(ocMetric)
+	for _, timeseries := range ocMetric.GetTimeseries() {
+		startTimestamp := timeseries.GetStartTimestamp()
```

Comment:
This can be converted once for all points in this timeseries

Reply:
done